src/librawrtc/sctp_transport.c

```diff
@@ -2241,6 +2222,45 @@ enum rawrtc_code rawrtc_sctp_transport_create(
 }
 
 /*
+ * Replace the underlying DTLS transport.
+ */
+enum rawrtc_code rawrtc_sctp_transport_set_transport(
+        struct rawrtc_sctp_transport* const transport,
+        struct rawrtc_dtls_transport* const dtls_transport // referenced
+) {
+    enum rawrtc_code error;
+
+    // Check arguments
+    if (!transport || !dtls_transport) {
+        return RAWRTC_CODE_INVALID_ARGUMENT;
+    }
+
+    // Check DTLS transport state
+    if (dtls_transport->state == RAWRTC_DTLS_TRANSPORT_STATE_CLOSED
+        || dtls_transport->state == RAWRTC_DTLS_TRANSPORT_STATE_FAILED) {
+        return RAWRTC_CODE_INVALID_STATE;
+    }
+
+    // Attach to DTLS transport
+    // Note: The function checks if a data transport is already registered for us
+    DEBUG_PRINTF("Attaching as data transport\n");
+    error = rawrtc_dtls_transport_set_data_transport(
```

Comment:
It should be possible to add multiple SCTP transports. *data_transport* transport is also incorrect, it should be *sctp_transport*. This should be changed once we have proper multiplexing.